galsim/wfirst/wfirst_detectors.py

```diff
@@ -122,6 +154,14 @@ def allDetectorEffects(img, rng=None, exptime=None):
     dark_noise = galsim.DeviateNoise(galsim.PoissonDeviate(rng, dark_current))
     img.addNoise(dark_noise)
 
+    # Persistence (use WFIRST coefficients)
+    applyPersistence(img,prev_exposures)
+
+    # Update the 'prev_exposures' queue
+    if len(prev_exposures)>len(galsim.wfirst.persistence_coefficients):
```

Comment:
This would have been a pretty harmless bug, but thanks for catching it
anyway.

On Tue, Apr 26, 2016 at 5:26 AM, Mike Jarvis <notifications@github.com>
wrote:

> In galsim/wfirst/wfirst_detectors.py
> <https://github.com/GalSim-developers/GalSim/pull/710#discussion_r61054317>
> :
>
> > @@ -122,6 +154,14 @@ def allDetectorEffects(img, rng=None, exptime=None):
> >      dark_noise = galsim.DeviateNoise(galsim.PoissonDeviate(rng, dark_current))
> >      img.addNoise(dark_noise)
> >
> > +    # Persistence (use WFIRST coefficients)
> > +    applyPersistence(img,prev_exposures)
> > +
> > +    # Update the 'prev_exposures' queue
> > +    if len(prev_exposures)>len(galsim.wfirst.persistence_coefficients):
>
> I'm pretty sure this is wrong. It should be >=. Or you can leave it like
> this, but move the insert command to happen before this check.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GalSim-developers/GalSim/pull/710/files/ab9d9f74f9bb2d3de9a54059daec1ac56e23cd7c#r61054317>
>



-- 
Regards

Arun Kannawadi Jayaraman
PhD candidate, Dept. of Physics,
Carnegie Mellon University,
Pittsburgh, PA 15213